polyply/src/assign_chirality.py

```diff
@@ -0,0 +1,216 @@
+    center:
+       node key of the center
+    max_depth:
+       maximum number of edges for neighbour search
+
+    Returns
+    -------
+    dict
+    """
+    masses = [(molecule.nodes[node]['mass'], node) for node in neighbours]
+    priority = {}
+    degree = 1
+    priorities = [4, 3, 2, 1]
+    while True:
+
+        if degree == max_depth:
+            msg = ("When determining priority a tie is "
+                   "remains unresolved after looking at "
+                   "the {} degree neighbours. Increase "
+                   "depth of search if you are sure the "
+                   "molecule is chiral.")
+            raise MaxIterationError(msg.format(degree))
+
+        masses.sort()
+        unique = _unique_masses(masses)
+        idx = 0
+        remove = []
+
+        for mass, node in masses:
+            if mass in unique:
+                mass, node = masses[idx]
+                priority[node] = priorities[idx]
+                remove.append(((mass, node), priorities[idx]))
+            idx += 1
+
+        if all([key in priority for key in priorities]):
+            status = True
+            break
+        else:
+            for entry in remove:
+                masses.remove(entry[0])
+                priorities.remove(entry[1])
+
+        idx = 0
+        for mass, node in masses:
+            mass = 0
+            neighbours, paths = neighborhood(molecule, node,
+                                             degree, min_length=degree+1,
+                                             not_cross=[center], return_paths=True)
+            for neigh in neighbours:
+                if degree == 1:
+                    edge = (node, neigh)
+                else:
+                    edge = (neigh, paths[neigh][-2])
+
+                bond_order = molecule.edges[edge]["bond_order"]
+                mass += molecule.nodes[neigh]["mass"] * bond_order
+
+            masses[idx] = (mass, node)
+            idx += 1
+        degree += 1
+        if masses and all([mass[0] == 0 for mass in masses]):
+            status = False
+            break
+
+    return status, priority
+
+
+def tag_chiral_centers(molecule, center_atom="C", priorities={}):
+    """
+    Given a `molecule` identify all chiral centers
+    and determine the priority of the subsituents
+    according the CIP rules. Centers are defined
+    as nodes of degree 4 and having the atomname
+    defined in `center_atom`.
+
+    Parameters:
+    ------------
+    molecule: vermouth.molecule
+    center_atom: string
+
+    Returns:
+    -------
+    nx.Graph
+    """
+    idx = 0
+    nx.set_node_attributes(molecule, False, "chiral")
+    centers = []
+    for node, degree in molecule.degree:
+        if degree == 4 and molecule.nodes[node]["atomname"].startswith(center_atom):
+            neighbours = molecule.neighbors(node)
+
+            if molecule.node[node]["atomname"] in skip_priority:
+                is_chiral = True
+                priority = 
```

Comment:
@pckroon at this place I would like to have the following functionality: If my center is in the predefined list (skip_priority) all neighbors for this and all other centers with same atom name that follow, should be ordered the same way. But I can't think of a clever way to do it. Basically the problem arises when the chiral center stretches two residues. In that case you can have the atom-name multiple times so you can't just make a dict of atomnames. Any clever idea how to deal with this?

Reply:
I'm not sure what you mean by this. Could you put an example?
How about using the tuple (atomname, resid) instead of just atomname?